### DISCUSSION TITLE
Can we serve mbtiles as .png using tileserver-gl so that leaflet.js can pick it up without using protobuf?

### DISCUSSION THREAD

**Discussion by AIBappa** [maptiler/tileserver-gl#1417] [Q&A] [answered]
Hi,
I am trying to serve the zurich.mbtiles file as a raster .png output from the tileserver-gl. Using the following configuration in config.json.

    "options": {
      "paths": {
        "root": "",
        "styles": "styles",
        "mbtiles": "mbtiles",
        "pmtiles": "pmtiles"
      },
        "formatOptions": {
            "png": {
                "palette": true,
                "colors": 4
                }
    },
      "maxScaleFactor": 3,
      "maxSize": 2048,
      "pbfAlias": "pbf",
      "serveAllFonts": true,
      "serveAllStyles": true,
      "serveStaticMaps": true,
      "allowRemoteMarkerIcons": true,
      "allowInlineMarkerImages": true,
      "staticAttributionText": "© OpenMapTiles  © OpenStreetMaps",
      "tileMargin": 0
    },
    "styles": {
      "maptiler-basic-gl-style": {
        "style": "style1.json",
        "serve_rendered": true,
        "tilejson": {
          "format": "png"
        }
      }
    },
    "data": {
      "zurich-vector": {
        "mbtiles": "zurich.mbtiles",
        "type": "raster"
      }
    }
  }

It is not working though. I can see the vector output in .pbf but not .png.
Any ideas?
Need leaflet.js to pick this up as a layer without resorting to using protobuf.

**Comment by acalcutt** [comment, marked as answer]
That image output option only applies to rendered tiles under 'styles', not data sources. So it would apply to the 'maptiler-basic-gl-style' in your config. 

For example, if you look at my styles at https://tiles.wifidb.net/ , if you click "XYZ" under a style it will show the url for rendered images right under. TileJSON also uses this image endpoint (which gets used in raster)
![image](https://github.com/user-attachments/assets/15678bba-6a15-49b6-9914-9796dff16557)

**Reply by AIBappa** [reply]
Thanks very much for elucidating this aspect about styles. Looking to update the style.json accordingly.

**Reply by acalcutt** [reply]
It our raster viewer I made this code to read the TileJSON url ( https://github.com/maptiler/tileserver-gl/blob/master/public/templates/viewer.tmpl#L82-L115 ) . and load it into leaflet.

**Reply by acalcutt** [reply]
but yes, modify your style as you said so it looks the way you want in the 'raster' view, then use the png endpoint mentioned and you should be able to use that with leaflet.

I have had styles that were just a single raster layer, but since you are using vector it takes a bit more styling.
